lib/galaxy/managers/workflows.py

```diff
@@ -1472,7 +1478,7 @@ def _workflow_to_dict_export(self, trans, stored=None, workflow=None, internal=F
                 "tool_version": module.get_version() if allow_upgrade else step.tool_version,
                 "name": module.get_name(),
                 "tool_state": json.dumps(tool_state),
-                "errors": module.get_errors(),
+                "errors": errors or module.get_errors(),
```

Comment:
I suppose a user would like to see the errors from `module.get_errors()` also when `errors` is not empty, i.e. we could merge them?
But currently `get_errors()` can return `None`, a list or a string, depending on the module type, which is not great in itself, so I suppose we need to decide what format this should have.

Reply:
errors will contain the upgrade messages, I don't think there's a need to merge them ?

Reply:
Then maybe put them in a "messages" key?

Reply:
Then I have to redo the editor and potentially other things. I don't disagree that this should be modernized, but it doesn't seem like the right moment ?

Reply:
And sorry, it's not messages, these are actually errors, for the current state of the module.

Reply:
Does the editor work with something that can be None, list, string or dictionary?
Anyway, my main concern was about losing the errors from `module.get_errors()` when `errors` is not empty.